capi-adoption/README.md

```diff
@@ -0,0 +1,159 @@
+
+We should have a certain level of defaulting for a set of versions we know to work well together, but we must not restrict customers from leveraging the flexibility of Cluster API.
+
+When we have already agreed to perform automatic minor and patch upgrades, we can equally just adopt the upstream workflow and update management cluster controllers in-place.
+
+## Proposal for alternative approach to provide Giant Swarm platform with upstream Cluster API
+
+To pick up the key points from current problems and to fully align with upstream in order to simplify our daily work and allow customers more flexibility while improving stability, here are proposed steps forward.
+
+### Let go of releases and multiple versions in same management cluster
+
+Currently the major source of friction with upstream and the source of complexity on our platform is coming from immutable releases and the need to run multiple versions of everything in the same Kubernetes cluster.
+
+Since the upstream Cluster API drastically changes the way how things are configured, we should leverage it, not work against it. Even though we have managed to contribute code upstream to support our use case, the community sees these contributions as workarounds to avoid blocking us. But we’d be always going in a different direction, increasing the effort and cost to fully implement CAPI and be aligned with the project.
+There are less risks for failure when different parts are managed separately and when active change of resources is always a direct consequence of active change on CR[s]. Management cluster operators can be upgraded when there are fixes, but they must not automatically change any of the reconciled resources. When there is a hot security fix for Kubernetes, it’s enough to update only the Kubernetes version. When workload cluster applications need updating, customers can perform it at their will via our App management infrastructure.
+
+Instead of strictly trying to keep control of release components, we should have sane defaults for CR templating. This would be our way to indicate “we have tested these components in this combination and found them to be rock solid, use them”.
+Focusing on templating with sane defaults rather than enforcing certain values would also mean getting rid of admission controllers that we currently use to validate and default CR content, simplifying the Kubernetes configuration.
+
+Allowing customers to customize those sane defaults would ease the testing of new combinations. For example, since upgrading Kubernetes to a new version would be only a matter of changing the required CRs, customers can test their workloads running on the new Kubernetes version on the cluster they decide at their own pace. And they would be testing that change in isolation. There would be no changes in the infrastructure resources or the installed apps.
```

Comment:
Would it make sense to get some sort of feedback loop in place, so that when customers do test a certain combo and it works, we can then know this and somehow document this for the use of other customers?

Reply:
Better late (comment/reply) than never :grimacing: Feedback loop as in customers giving us feedback how the testing went when they used certain combination of Kubernetes version and other component versions?

I would agree very much that we would need some sort of insight into issues that are connected to a combination of components running in a cluster (what we currently have as a "release" concept), e.g. we should be able to see (or easily discover) that running `Kubernetes 1.X.Y`, `Flatcar X.Y.Z` and `operator X v1.1.1` causes issues `A`, `B` and `C` to happen more frequently (or never).